src/include/sof/audio/host_copier.h

```diff
@@ -117,7 +118,67 @@ static inline int host_common_copy(struct host_data *hd, struct comp_dev *dev, c
 {
 	return hd->copy(hd, dev, cb);
 }
-void host_common_update(struct host_data *hd, struct comp_dev *dev, uint32_t bytes);
+
+static inline void host_common_update(struct host_data *hd, struct comp_dev *dev, uint32_t bytes)
```

Comment:
this one seems too big for a header to me. And it is called from multiple locations: e.g. from host-zephyr.c and from copier_host.c, so it will increase code size. We cannot inline everything, done in `.copy()` - we will have function calls and we should have them. We could try and inline "everything" effectively creating a single huge function, but that doesn't seem like a good idea to me...